tasks/code_sync_status.rb

```diff
@@ -0,0 +1,93 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+
+# Puppet Task Name: code_sync_status
+require 'net/https'
+require 'uri'
+require 'json'
+require 'puppet'
+
+# CodeSyncStatus task class
+class CodeSyncStatus
+  def initialize(params)
+    @params = params
+  end
+
+  def execute!
+    puts syncstatus.to_json
+  end
+
+  def https
+    https = Net::HTTP.new('localhost', '8140')
+    https.use_ssl = true
+    https.cert = @cert ||= OpenSSL::X509::Certificate.new(File.read(Puppet.settings[:hostcert]))
+    https.key = @key ||= OpenSSL::PKey::RSA.new(File.read(Puppet.settings[:hostprivkey]))
+    https.verify_mode = OpenSSL::SSL::VERIFY_NONE
+    https
+  end
+
+  def apistatus
+    status = https
+    # Only debug level includes code sync details
+    status_request = Net::HTTP::Get.new('/status/v1/services?level=debug')
+    JSON.parse(status.request(status_request).body)
+  end
+
+  def checkenvironmentlist(environments, requestedenvironments)
```

Comment:
Based on the method name convention, should we change this method name?

```suggestion
  def check_environment_list(environments, request_environments)
```

Also a cosmetic thing, there is a typo in the `edenvironments` param :)